### DISCUSSION TITLE
Create 'gallery' of images or videos uploaded together and display them as a single message/event

### DISCUSSION THREAD

**Discussion by Nesaijn** [element-hq/element-meta#590] [Ideas]
#### What would you like to do?
I want the media that I have selected to send at the same time to appear as a kind of gallery in the chat, like grouped together. Clicking on the grouped media would get me to a focused gallery view with only the media.

#### Why would you like to do it?
When sending many files they take up a lot of space and you would have to scroll a lot to get past them or find the picture you are searching. With them being grouped you also can see the messages before easily.
Also while the media uploading one by one it is common that people already start commenting on the media that they are seeing. But more are being sent after the messages so that the they are getting buried between the media making them harder to find and making us to also scroll more to search for the messages. Grouping them would lead to a better experience when sending batches of media while having a cleaner chat history.

#### How would you like to achieve it?
An idea would be to maybe add a key to the data of media files, which are sent in a batch, with an id to identify which media to group. Also adding to it the number of media files the batch contains so that the client knows the number of placeholders to create while the media is being uploaded.

### Are you willing to provide a PR?

No

**Comment by hoka99** [comment]
Whatsapp does this really well .. then you can just click a single download button and get ALL of the media instead of having to get each image individually

**Reply by TemoCrevlis** [reply]
On desktop there isn't currently a way to even cycle through media in a room.
You have to open, close and scroll to, for each pic/video.
Giving the feature to do that would be a good start toward a gallery, I think.
